src/vcpkg/base/downloads.cpp

```diff
@@ -28,7 +28,45 @@ namespace vcpkg
         return input;
     }
 
+    bool operator==(const ProxyCredentials& lhs, const ProxyCredentials& rhs)
+    {
+        return lhs.username == rhs.username && lhs.password == rhs.password;
+    }
+
+    bool operator==(const ProxyUrlParts& lhs, const ProxyUrlParts& rhs)
+    {
+        return lhs.host == rhs.host && lhs.credentials == rhs.credentials;
+    }
+
+    ProxyUrlParts parse_proxy_url(const std::wstring& url)
+    {
+        constexpr auto npos = std::wstring::npos;
+        auto scheme_separator_pos = url.find(L"://");
+        std::wstring scheme;
+        std::wstring_view remaining_parts = url;
```

Comment:
Please use StringView rather than std::wstring_view as we don't want to add a dependency on a new C++17 feature in order to have this. (Note that this would also require the char* rather than wchar_t* suggestion below)